### PR TITLE
Re-record Batch tests

### PR DESCRIPTION
**Pull request by matthchr**
- Also label tests as "CheckIn" category.

## Description

See title

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by matthchr** [comment]
This PR supplants PR #6249

File: src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/ScenarioTestHelpers.cs [hunk 1]
```diff
@@ -507,7 +507,7 @@ public static void WaitForIdleComputeNode(BatchController controller, BatchAccou
                 Select = "id,state"
             };
 
-            DateTime timeout = DateTime.Now.AddMinutes(5);
+            DateTime timeout = DateTime.Now.AddMinutes(10);
```
Comment (maddieclayton) [inline src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/ScenarioTestHelpers.cs]: Can we change this to 3 hours?

Comment (matthchr) [reply src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/ScenarioTestHelpers.cs]: I wanted to see where the tests failed in CI, and then fix up all those places, as I am not sure this is the only place we wait.

I will amend the PR to wait a longer time once I've seen the initial results though, yes.

**Comment by matthchr** [comment]
This fixes issue: #5006

**Comment by matthchr** [comment]
It looks like the test pass in Travis failed because of this error:
```
[xUnit.net 00:00:20.4340168]     Microsoft.Azure.Commands.Batch.Test.ScenarioTests.CertificateTests.TestCertificateCrudOperations [FAIL]
Failed   Microsoft.Azure.Commands.Batch.Test.ScenarioTests.CertificateTests.TestCertificateCrudOperations
Error Message:
 System.Management.Automation.CmdletInvocationException : Exception calling ".ctor" with "1" argument(s): "error:2006D080:BIO routines:BIO_new_file:no such file"
---- System.Management.Automation.MethodInvocationException : Exception calling ".ctor" with "1" argument(s): "error:2006D080:BIO routines:BIO_new_file:no such file"
-------- Interop+Crypto+OpenSslCryptographicException : error:2006D080:BIO routines:BIO_new_file:no such file
Stack Trace:
   at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
   at System.Management.Automation.Runspaces.Pipeline.Invoke()
   at System.Management.Automation.PowerShell.Worker.ConstructPipelineAndDoWork(Runspace rs, Boolean performSyncInvoke)
   at System.Management.Automation.PowerShell.Worker.CreateRunspaceIfNeededAndDoWork(Runspace rsToUse, Boolean isSync)
   at System.Management.Automation.PowerShell.CoreInvokeHelper[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.CoreInvoke[TInput,TOutput](PSDataCollection`1 input, PSDataCollection`1 output, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.CoreInvoke[TOutput](IEnumerable input, PSDataCollection`1 output, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.Invoke(IEnumerable input, PSInvocationSettings settings)
   at System.Management.Automation.PowerShell.Invoke()
   at Microsoft.WindowsAzure.Commands.ScenarioTest.EnvironmentSetupHelper.RunPowerShellTest(String[] scripts) in /home/travis/build/Azure/azure-powershell/src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/EnvironmentSetupHelper.cs:line 638
   at Microsoft.Azure.Commands.Batch.Test.ScenarioTests.BatchController.RunPsTestWorkflow(Func`1 scriptBuilder, Action initialize, Action cleanup, String callingClassType, String mockName) in /home/travis/build/Azure/azure-powershell/src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/BatchController.cs:line 112
   at Microsoft.Azure.Commands.Batch.Test.ScenarioTests.BatchController.RunPsTest(String[] scripts) in /home/travis/build/Azure/azure-powershell/src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/BatchController.cs:line 55
   at Microsoft.Azure.Commands.Batch.Test.ScenarioTests.CertificateTests.TestCertificateCrudOperations() in /home/travis/build/Azure/azure-powershell/src/ResourceManager/AzureBatch/Commands.Batch.Test/ScenarioTests/CertificateTests.cs:line 34
----- Inner Stack Trace -----
   at System.Management.Automation.DotNetAdapter.AuxiliaryConstructorInvoke(MethodInformation methodInformation, Object[] arguments, Object[] originalArguments)
   at System.Management.Automation.DotNetAdapter.InvokeResolvedConstructor(MethodInformation bestMethod, Object[] newArguments, Object[] arguments)
   at System.Management.Automation.DotNetAdapter.ConstructorInvokeDotNet(Type type, ConstructorInfo[] constructors, Object[] arguments)
   at Microsoft.PowerShell.Commands.NewObjectCommand.CallConstructor(Type type, ConstructorInfo[] constructors, Object[] args)
----- Inner Stack Trace -----
   at Interop.Crypto.CheckValidOpenSslHandle(SafeHandle handle)
   at Internal.Cryptography.Pal.CertificatePal.FromFile(String fileName, SafePasswordHandle password, X509KeyStorageFlags keyStorageFlags)
   at System.Security.Cryptography.X509Certificates.X509Certificate..ctor(String fileName, String password, X509KeyStorageFlags keyStorageFlags)
   at System.Security.Cryptography.X509Certificates.X509Certificate2..ctor(String fileName)
```

I _think_ it's complaining because we have a file `Resources\BatchTestCert01.cer` that it is trying to use but can't find?

I see this issue: https://github.com/dotnet/corefx/issues/11042 suggests the same as above (file not found).

But this same test passes in Jenkins. Do we explicitly exclude .cer files from the travis run somehow?

**Comment by maddieclayton** [comment]
@MiYanni Can you take a look at this?

**Comment by MiYanni** [comment]
@matthchr I just had the same issue for another service's tests. The fix here:
https://github.com/Azure/azure-powershell/pull/6281/commits/4ab7861d762440d7ad3b55ff515c1b6e7e898793

Basically, the path resolution seems to vary depending on if you are running the test via the console runner or the Visual Studio runner. Just do the logic I did in **green** where you have both potential paths and check to see if one exists; otherwise, use the other path.

**Comment by maddieclayton** [comment]
on demand (to ensure all batch tests are successfully moved): https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/920/

**Comment by maddieclayton** [comment]
@matthchr @MiYanni Looks like the test is still failing in travis.

**Comment by matthchr** [comment]
@MiYanni - Can you give specific guidance on how to do your fix in a Powershell context rather than a C# one?

I tried:
```
    $certPathVs = $localDir + "\Resources\BatchTestCert01.cer"
    $certPathConsole = [System.IO.Path]::GetDirectoryName($PSScriptRoot) + "\Resources\BatchTestCert01.cer"
    $certPath = if (Test-Path $certPathVs -PathType Leaf) { $certPathVs } Else { $certPathConsole }
    $x509cert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2 -ArgumentList $certPath
```

But that doesn't seem to work.

I assume the issue is the `certPathConsole` path... what is the directory structure in Travis?

**Comment by MiYanni** [comment]
@matthchr Oh. Your issue isn't that the directory is wrong. Just put:
```powershell
$x509cert = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2 -ArgumentList './Resources/BatchTestCert01.cer'
```

### **You can't use backslashes in Linux.**

**Comment by matthchr** [comment]
There we go, Travis passes now. Feel free to call another on-demand build to make sure things are looking okay.

**Comment by maddieclayton** [comment]
On demand will be fine as I confirmed all batch tests were moved.  Merging now! I'll close the flaky test issue now, but let you know if it ever shows back up.

Thanks Matt!  We really appreciate your work fixing these tests.

**Pull request merged**